Description: Fix time_t value formatting in k8s support
Author: Zixing Liu <zixing.liu@canonical.com>
Forwarded: no
Last-Update: 2024-03-22

--- rsyslog-8.2312.0.orig/contrib/mmkubernetes/mmkubernetes.c
+++ rsyslog-8.2312.0/contrib/mmkubernetes/mmkubernetes.c
@@ -1722,7 +1722,7 @@ queryKB(wrkrInstanceData_t *pWrkrData, c
 		now -= pWrkrData->pData->cache->lastBusyTime;
 		if (now < pWrkrData->pData->busyRetryInterval) {
 			LogMsg(0, RS_RET_RETRY, LOG_DEBUG,
-				"mmkubernetes: Waited [%ld] of [%d] seconds for the requested url [%s]\n",
+				"mmkubernetes: Waited [%" PRId64 "] of [%d] seconds for the requested url [%s]\n",
 				now, pWrkrData->pData->busyRetryInterval, url);
 			ABORT_FINALIZE(RS_RET_RETRY);
 		} else {
